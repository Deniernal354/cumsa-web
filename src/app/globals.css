@import "tailwindcss";

:root {
  --background: #f8fafc;
  --foreground: #0f172a;
  --surface: #ffffff;
  --muted: #64748b;
  --navy: #2a7487;
  --blue: #5aced8;
  --red: #c73221;
  --header-top: #123a66;
}

body {
  background: var(--background);
  color: var(--foreground);
  font-family: Arial, Helvetica, sans-serif;
}

a {
  text-underline-offset: 4px;
}

.link-cta {
  color: var(--blue);
  font-weight: 700;
  text-decoration: underline;
  text-underline-offset: 3px;
}

.link-cta:hover {
  color: var(--red);
}

.btn-cozy {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  padding: 0.5rem 1rem;
  border-radius: 0.6rem;
  font-weight: 600;
  font-size: 0.9rem;
  transition:
    transform 150ms ease,
    box-shadow 150ms ease,
    opacity 150ms ease;
}

.btn-cozy:hover {
  transform: translateY(-1px);
  box-shadow: 0 6px 16px rgba(0, 0, 0, 0.08);
}

.btn-primary {
  background: var(--blue);
  color: white;
}

.btn-outline {
  background: var(--surface);
  color: var(--foreground);
  border: 1px solid rgba(0, 0, 0, 0.12);
}

:focus-visible {
  outline: 2px solid color-mix(in oklab, var(--blue) 70%, white);
  outline-offset: 2px;
}

.container-base {
  margin-left: auto;
  margin-right: auto;
  max-width: 80rem;
  padding-left: 1rem;
  padding-right: 1rem;
}

@media (min-width: 640px) {
  .container-base {
    padding-left: 1.5rem;
    padding-right: 1.5rem;
  }
}

@media (min-width: 1024px) {
  .container-base {
    padding-left: 2rem;
    padding-right: 2rem;
  }
}

.transition-base {
  transition: all 200ms ease-out;
}

.card {
  background: var(--surface);
  border: 1px solid rgba(0, 0, 0, 0.12);
  border-radius: 0.75rem;
}

.title {
  color: var(--red);
}

.tint-blue {
  filter: invert(28%) sepia(88%) saturate(3612%) hue-rotate(211deg) brightness(93%) contrast(98%);
}

.logo-mask-blue {
  background-color: var(--blue);
  -webkit-mask-image: url("/CU-MSA-LOGO-blank-2.png");
  mask-image: url("/CU-MSA-LOGO-blank-2.png");
  -webkit-mask-repeat: no-repeat;
  mask-repeat: no-repeat;
  -webkit-mask-position: center;
  mask-position: center;
  -webkit-mask-size: contain;
  mask-size: contain;
  display: inline-block;
}
