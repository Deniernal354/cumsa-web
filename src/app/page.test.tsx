import { test, expect } from "@jest/globals";
import { render } from "@testing-library/react";
import Home from "./page";

test("home page snapshot", () => {
  const { asFragment } = render(<Home />);
  expect(asFragment()).toMatchSnapshot();
});
