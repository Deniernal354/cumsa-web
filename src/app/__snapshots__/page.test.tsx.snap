// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home page snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      data-testid="stub"
    />
    <section
      class="bg-white"
    >
      <div
        class="container-base py-8 flex justify-center"
      >
        <img
          alt="Bismillah banner"
          class="w-full max-w-4xl h-auto rounded-lg border border-black/10"
          height="200"
          src="/salam-1.png"
          width="900"
        />
      </div>
    </section>
    <section
      class="bg-[var(--surface)]"
    >
      <div
        class="container-base py-12"
      >
        <h2
          class="text-2xl font-bold text-center title"
        >
          Our Services
        </h2>
        <div
          class="mt-6 grid gap-6 md:grid-cols-3"
        >
          <div
            data-testid="stub"
          >
            <a
              class="no-underline"
              href="/services/prayer"
            >
              <h3
                class="font-semibold text-lg"
              >
                Prayers
              </h3>
              <p
                class="mt-2 text-gray-600"
              >
                Find prayer spaces on campus and Jummah updates.
              </p>
              <span
                class="mt-3 inline-block link-cta"
              >
                Click Here →
              </span>
            </a>
          </div>
          <div
            data-testid="stub"
          >
            <a
              class="no-underline"
              href="/services/equity"
            >
              <h3
                class="font-semibold text-lg"
              >
                Equity Services
              </h3>
              <p
                class="mt-2 text-gray-600"
              >
                Know your rights and how to request religious accommodations.
              </p>
              <span
                class="mt-3 inline-block link-cta"
              >
                Click Here →
              </span>
            </a>
          </div>
          <div
            data-testid="stub"
          >
            <a
              class="no-underline"
              href="/services/roommate"
            >
              <h3
                class="font-semibold text-lg"
              >
                Roommate Services
              </h3>
              <p
                class="mt-2 text-gray-600"
              >
                Need a place to stay? Search community listings.
              </p>
              <span
                class="mt-3 inline-block link-cta"
              >
                Click Here →
              </span>
            </a>
          </div>
        </div>
      </div>
    </section>
    <section>
      <div
        class="container-base py-12"
      >
        <h2
          class="text-2xl font-bold text-center title"
        >
          Our Purpose
        </h2>
        <div
          class="mt-6 grid gap-8 md:grid-cols-3"
        >
          <div
            class="card p-6"
          >
            <h3
              class="font-semibold"
            >
              Who We Are
            </h3>
            <p
              class="mt-2 text-gray-700"
            >
              We’re the campus hub for practicing and growing in faith, building an inclusive community for students.
            </p>
          </div>
          <div
            class="card p-6"
          >
            <h3
              class="font-semibold"
            >
              What We Do
            </h3>
            <p
              class="mt-2 text-gray-700"
            >
              We represent Muslim students, organize services and events, and encourage civic engagement and learning.
            </p>
          </div>
          <div
            class="card p-6"
          >
            <h3
              class="font-semibold"
            >
              Who We Serve
            </h3>
            <p
              class="mt-2 text-gray-700"
            >
              We support the Muslim student body at Carleton—our diverse family keeps growing every year.
            </p>
          </div>
        </div>
      </div>
    </section>
    <div
      data-testid="stub"
    />
  </div>
</DocumentFragment>
`;
